web-common-marlowe/src/Marlowe/Extended.purs

```diff
@@ -29,15 +29,45 @@ type ContractTemplate
     }
 
 type MetaData
-  = { contractName :: String
-    , contractType :: String
+  = { contractType :: ContractType
     , contractDescription :: String
     , roleDescriptions :: Map S.TokenName String
     , slotParameterDescriptions :: Map String String
     , valueParameterDescriptions :: Map String String
     , choiceDescriptions :: Map String String
     }
 
+data ContractType
+  = ES
+  | EC
+  | ZC
+  | CB
+  | S
+  | CD
+
+derive instance eqContractType :: Eq ContractType
+
+instance showContractType :: Show ContractType where
```

Comment:
We add more meaning to the `Show` typeclass than we should. I'd use a `contractAcronym` function